### PR TITLE
Invalidate icache before letting the CWCheat engine change memory. Might fix JIT-related cheat crashes.

### PR DESCRIPTION
**Pull request by hrydgard**
Please test if you use CWCheat and experience related crashes (I don't really :)

### CODE REVIEW

File: Core/CwCheat.cpp [hunk 1]
```diff
@@ -280,6 +282,12 @@ std::vector<std::string> CWCheatEngine::GetCodesList() { //Reads the entire chea
 	return codesList;
 }
 
+void CWCheatEngine::InvalidateICache(u32 addr, int size) {
+	if (MIPSComp::jit) {
+		MIPSComp::jit->GetBlockCache()->InvalidateICache(addr & ~3, 4);
```
Comment (unknownbrackets) [inline Core/CwCheat.cpp]: Should use the `size` parameter that was passed rather than 4.

-[Unknown]

Comment (hrydgard) [inline Core/CwCheat.cpp]: Eh, doh. Thanks. Silly copy paste error

**Comment by LunaMoo** [comment]
Checked it with that broken code which Melkor sent me and this solves it without having to clean up the cheat itself. It also fixed joker codes that directly changed asm, so I don't think there are any bothersome cheat engine problems left after this get's merged.:)
 I was affraid this would affect performance, but doesn't seem to do that in any measurable way and even if it would, people can now set refresh rate for cheats to a much slower one.

**Comment by hrydgard** [comment]
@LunaMoo , thanks for testing!

**Comment by AizerMortenort** [comment]
Tried it with several codes and mods - no freeze. 
Clearing cache ?

**Pull request merged**